roomserver/storage/shared/storage.go

```diff
@@ -468,6 +468,11 @@ func (d *Database) StoreEvent(
 		if !isRejected { // ignore rejected redaction events
 			redactionEvent, redactedEventID, err = d.handleRedactions(ctx, txn, eventNID, event)
 		}
+		for _, prev := range event.PrevEvents() {
+			if err = d.PrevEventsTable.InsertPreviousEvent(ctx, txn, prev.EventID, prev.EventSHA256, eventNID); err != nil {
```

Comment:
It's probably not safe to do this because of the comment:
```
// Insert an entry into the previous_events table.
// If there is already an entry indicating that an event references that previous event then
// add the event NID to the list to indicate that this event references that previous event as well.
// This should only be modified while holding a "FOR UPDATE" lock on the row in the rooms table for this room.
// The lock is necessary to avoid data races when checking whether an event is already referenced by another event.
```

Reply:
Hmm, good point, I can change this to use the latest rooms updater although that may still not do the right thing on SQLite anyway.